File: 3010.divide-an-array-into-subarrays-with-minimum-cost-i/solution.go
func minimumCost(nums []int) int {
    min1, min2 := math.MaxInt, math.MaxInt
    for _, n := range nums[1:] {
        if n < min1 {
            min1, min2 = n, min1
        } else if n < min2 {
            min2 = n
        }
    }
    return nums[0] + min1 + min2
}

File: 3010.divide-an-array-into-subarrays-with-minimum-cost-i/solution.kt
class Solution {
    fun minimumCost(nums: IntArray): Int {
        var min1 = Int.MAX_VALUE
        var min2 = Int.MAX_VALUE
        for (i in 1..nums.lastIndex) {
            val n = nums[i]
            if (n < min1) {
                min2 = min1
                min1 = n
            } else if (n < min2) {
                min2 = n
            }
        }
        return nums.first() + min1 + min2
    }
}